src/animation/use-value-animation-controls.ts

```diff
@@ -42,24 +42,27 @@ export function useValueAnimationControls<P>(
         controls.setProps(props)
         controls.setVariants(variants)
         controls.setDefaultTransition(transition)
+    }
 
+    // We have to subscribe to the parent controls within a useEffect rather than during render,
+    // as
+    useEffect(() => {
         if (subscribeToParentControls && parentControls) {
             parentControls.addChild(controls)
         }
-    }
+    })
```

Comment:
I'm wondering - if you `resetChildren` in render, but addChild in an effect - thing won't be animatable (and maybe even animation in progress will get canceled?) between those 2 points in time, it doesn't sound right as from the user's perspective this is not observable. I know actually trying to start an animation in between of those is very unlikely - but theoretically not impossible.